contracts/src/TeleporterTokenSource.sol

```diff
@@ -71,11 +101,69 @@ abstract contract TeleporterTokenSource is
         feeTokenAddress = feeTokenAddress_;
     }
 
+    function _registerDestination(
+        bytes32 destinationBlockchainID,
+        address destinationBridgeAddress,
+        uint256 initialReserveImbalance,
+        uint256 tokenMultiplier,
+        bool multiplyOnSend
+    ) internal {
+        require(
+            destinationBlockchainID != bytes32(0),
+            "TeleporterTokenSource: zero destination blockchain ID"
+        );
+        require(
+            destinationBlockchainID != blockchainID,
+            "TeleporterTokenSource: cannot register bridge on same chain"
+        );
+        require(
+            destinationBridgeAddress != address(0),
+            "TeleporterTokenSource: zero destination bridge address"
+        );
+        require(
+            tokenMultiplier > 0 && tokenMultiplier < 1e18,
+            "TeleporterTokenSource: invalid token multiplier"
+        );
+        require(
+            !registeredDestinations[destinationBlockchainID][destinationBridgeAddress].registered,
+            "TeleporterTokenSource: destination already registered"
+        );
+
+        // Calculate the collateral needed in source token denomination.
+        uint256 collateralNeeded = TokenScalingUtils.removeTokenScale(
+            tokenMultiplier, multiplyOnSend, initialReserveImbalance
+        );
+
+        // Round up the collateral needed by 1 in the case that `multiplyOnSend` is true and
+        // `initialReserveImbalance` is not divisible by the `tokenMultiplier` to
+        // ensure that the full amount is account for.
+        if (multiplyOnSend && initialReserveImbalance % tokenMultiplier != 0) {
+            collateralNeeded += 1;
+        }
```

Comment:
I think this makes sense if the destination is scaling the `initialReserveImbalance` value it sends to register, but right now we're not doing that (but I think we should)

Reply:
in the above lines, we remove the token scaling of the destination's `initialReseveImbalance` and calculate collateral in source token amount